frontend/src/app/core/state/principals/user.model.ts

```diff
@@ -0,0 +1,31 @@
+import { ID } from '@datorama/akita';
+import { HalResourceLink, HalResourceLinks } from 'core-app/core/state/hal-resource';
+
+export interface UserHalResourceLinks extends HalResourceLinks {
```

Comment:
I actually think the `HalResourceLinks` definition is wrong. It should only contain the `self` link instead of a record of links. That is very similar to an index type and will be painful down the road. We can then extend the type with a `CustomFieldLinks` down the line that has a record interface.

Reply:
Yeah I thought the type wasn't correct, however I'm not sure  I understand your proposed change. Like this?

```
Interface HalResourceLinks {
  self:HalResourceLink;
}
```